Persistent memory support for hybrid ISOs

Nowa-Ammerlaan opened the pull request:
This adds a new option for use in the spec files which will toggle the creation of an extra partition in the hybrid ISO image. The partition file system is XFS only (for now), and is intended for use as persistent storage when the ISO is `dd`ed to e.g. an USB drive. The extra partition has no effect when the ISO is burned to an optical medium as partitioning and writing back is not really supported there.

There are a couple of things still to discuss (either now or in a follow-up PR):
- Do we want the file system type to be variable, i.e. add another setting so we can choose something other then XFS?
- Can we change the name of the partition (currently `Appended3`)? Do we want to? The `xorriso` manual is unclear where the name comes from.
- Do we want to automatically mount this partition at boot (i.e. add it to the `fstab`)? If so where should we mount it?

See-also: https://gist.github.com/immolo/16de4fe7907991878585eb2c56e31f27

CC @immolo

Nowa-Ammerlaan committed:
create-iso.sh: add support for creating an extra partition

- second argument (if specified) will control the size of this
partition.

- We give the partition the GUID of a generic Linux data partition,
when translated to MBR this also maps to a Linux partition
automatically.

- The arguments we are adding are options for xorriso, not for
grub-mkrescue. Grub-mkrescue passes on whatever we give it extra.

- XFS is hardcoded for now, we can extend this in the future,
possibly by adding support for a third argument.

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

Nowa-Ammerlaan committed:
targets/*/controller.sh: pass on all extra arguments to create-iso.sh

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

Nowa-Ammerlaan committed:
catalyst/*: add new option iso_extra_partition

When specified in the spec, this will instruct create-iso.sh to
create an extra XFS partition of the specified size.

The intended use is having a partition available for persistent
storage when the created hybrid ISO image is written to e.g. a USB.

When the hybrid ISO is burned to an optical medium the extra
partition is ignored, along with all other MBR data.

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

Nowa-Ammerlaan committed:
doc: document new iso_extra_partition option

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

immolo commented:
This is awesome, thank you <3  I have test images I can share in IRC however it works extremely well in the tests I have run.

> Can we change the name of the partition?

Inside the DEs I have tested it just shows up $PART-SIZE as the name to users and it's as good as name as any to use a LABEL name for mounting. I vote leave it as it is.

> Do we want to automatically mount?

Seems logically, I'm currently using [fsscript](https://github.com/gentoo/releng/blob/master/releases/specs/amd64/livegui/files/fsscript-stage2.sh) to do the same task. As for name `/mnt/storage`?

Nowa-Ammerlaan committed:
functions.sh: pass on arguments in exec_in_chroot function

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

Nowa-Ammerlaan committed:
livecdfs-update.sh: register data storage partition in fstab

the mount point is passed as the first argument to this script. If
none then nothing is added to the fstab.

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

Nowa-Ammerlaan committed:
controller.sh: pass on arguments when calling livecdfs-update.sh

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

Nowa-Ammerlaan committed:
stagebase.py: register data storage partition in livecd fstab

Signed-off-by: Nowa Ammerlaan <nowa@gentoo.org>

Nowa-Ammerlaan commented:
> Seems logically, I'm currently using [fsscript](https://github.com/gentoo/releng/blob/master/releases/specs/amd64/livegui/files/fsscript-stage2.sh) to do the same task. As for name `/mnt/storage`?

Done! I made the mounting point an argument for the script, so we can more easily turn it into a setting for the spec file if we want to in the future.

akhuettel commented:
This looks great. Pulling it in - and sorry for the long waiting time. In case of doubt poke me in IRC in the future.
